### PR TITLE
TOSDR data fixes

### PR DESCRIPTION
**Pull request by andrey-p**
**Reviewer:** @jdorweiler 

**CC:** @dharb @russellholt 

## Description:

This bunches several TOSDR fixes in the same PR (sorry). I'll add more of a description inline.

## Steps to test this PR:

1. Run `grunt execute:tosdr --browser=chrome --type=dev`
2. Try the sites I've mentioned in the inline comments

## Automated tests:
- [ ] Unit tests
- [ ] Integration tests

###### Reviewer Checklist:
- [ ] **Ensure the PR solves the problem**
- [ ] **Review every line of code**
- [ ] **Ensure the PR does no harm by testing the changes thoroughly**
- [ ] **Get help if you're uncomfortable with any of the above!**
- [ ] Determine if there are any quick wins that improve the implementation

###### PR Author Checklist:
- [ ] Get advice or leverage existing code
- [ ] Agree on technical approach with reviewer (if the changes are nuanced)
- [ ] Ensure that there is a testing strategy (and documented non-automated tests)
- [ ] Ensure there is a documented monitoring strategy (if necessary)
- [ ] Consider systems implications

### CODE REVIEW

File: scripts/tosdr.js [hunk 1]
```diff
@@ -50,12 +50,27 @@ function getSitePoints (sites) {
...
+                addPoint(points, type, pointCase, score)
+        }
+
+        // we use class in our score but we may not have privacy-related reasons for it
+        // so show all available reasons instead
+        if (points.class &&
+                (!points.match.good || !points.match.good.length) &&
+                (!points.match.bad || !points.match.bad.length)) {
+            points.match.good = points.all.good
+            points.match.bad = points.all.bad
```
Comment (andrey-p) [inline scripts/tosdr.js]: There's an edge case where we get a class from TOSDR, but no privacy-related points. This falls back on displaying all TOSDR points so the user can see why we gave it that class. See e.g. seenthis.net:

screenshot from 2018-01-22 18-35-16

File: scripts/tosdr.js [hunk 2]
```diff
@@ -82,7 +97,9 @@ function addPoint(points, type, pointCase, score) {
     points['all'][type].push(pointCase)
     
     // is this a point we care about
-    if (topics[type].indexOf(pointCase) !== -1){
+    if (topics[type].indexOf(pointCase) !== -1 &&
+            // avoid adding duplicate points
+            points['match'][type].indexOf(pointCase) === -1){
```
Comment (andrey-p) [inline scripts/tosdr.js]: Avoids duplicate points, see https://github.com/duckduckgo/duckduckgo-privacy-extension/pull/73#issuecomment-359214171.

screenshot from 2018-01-22 18-36-41

File: scripts/tosdr-topics.json [hunk 3]
```diff
@@ -7,7 +7,8 @@
         "tracks you on other websites",
         "logs are kept forever",
         "they can use your content for all their existing and future services",
-        "your data is used for many purposes"
+        "your data is used for many purposes",
+        "device fingerprinting"
```
Comment (andrey-p) [inline scripts/tosdr-topics.json]: This was added by TOSDR in the past couple of days. I thought it's worth adding it here.

File: scripts/tosdr.js [hunk 4]
```diff
@@ -50,12 +50,27 @@ function getSitePoints (sites) {
...
             let point = pointsData[pointName]
             let pointCase = point.tosdr.case
+            let score = point.tosdr.score || 0
             if (!pointCase) continue
-            
+
+            // standardize case (some of them start with caps)
+            pointCase = pointCase.toLowerCase()
+            // standardize score (some of them come as strings)
+            score = parseInt(score, 10)
```
Comment (andrey-p) [inline scripts/tosdr.js]: There were a couple of inconsistencies that we need to deal with here. "device fingerprinting" comes from TOSDR with a capitalized "D", and I saw some string values for `score`, which led to them being concatenated below, e.g. "050".

**Comment by jdorweiler** [comment]
👍

**Pull request merged**